classes/Language.php

```diff
@@ -1338,54 +1346,31 @@ public static function updateMultilangFromClass($table, $className, $lang)
     }
 
     /**
-     * Language::installRtlStylesheets()
-     *
-     * @param bool $boTheme
-     * @param bool $foTheme
-     * @param string|null $themeName
-     * @param string|null $iso
-     * @param bool $install
-     * @param string|null $path
+     * Returns an RTL stylesheet processor instance
      *
-     * @throws \PrestaShop\PrestaShop\Core\Localization\RTL\Exception\GenerationException
-     * @throws Exception
+     * @return RtlStylesheetProcessor
      */
-    public static function installRtlStylesheets($boTheme = false, $foTheme = false, $themeName = null, $iso = null, $install = false, $path = null)
+    public static function getRtlStylesheetProcessor()
     {
-        if ($iso) {
-            $lang_pack = static::getLangDetails($iso);
-            if (!$lang_pack['is_rtl']) {
-                return;
-            }
-        }
-
-        $generator = new StylesheetGenerator();
-
-        // generate stylesheets for BO themes
-        if ($boTheme) {
-            if (defined('_PS_ADMIN_DIR_')) {
-                $adminDir = _PS_ADMIN_DIR_;
-            } else {
-                $adminDir = _PS_ROOT_DIR_.DIRECTORY_SEPARATOR.'admin';
-                $adminDir = (is_dir($adminDir)) ? $adminDir : ($adminDir.'-dev');
-            }
-
-            if (!is_dir($adminDir)) {
-                throw new Exception("Cannot generate BO themes: \"$adminDir\" is not a directory");
-            }
-
-            $generator->generateInDirectory($adminDir.DIRECTORY_SEPARATOR.'themes');
+        if (defined('_PS_ADMIN_DIR_')) {
+            $adminDir = _PS_ADMIN_DIR_;
+        } else {
+            $adminDir = _PS_ROOT_DIR_.DIRECTORY_SEPARATOR.'admin';
+            $adminDir = (is_dir($adminDir)) ? $adminDir : ($adminDir.'-dev');
         }
 
-        // generate stylesheets for BO themes
-        if ($foTheme) {
-            $frontDir = _PS_ROOT_DIR_.DIRECTORY_SEPARATOR.'themes'.DIRECTORY_SEPARATOR;
+        $themesDir = _PS_ROOT_DIR_.DIRECTORY_SEPARATOR.'themes';
 
-            $generator->generateInDirectory($frontDir.($themeName?$themeName:'classic'));
-        }
+        $processor = new RtlStylesheetProcessor(
+            $adminDir,
+            $themesDir,
+            array(
+                _PS_MODULE_DIR_.'gamification',
+                _PS_MODULE_DIR_.'welcome',
+                _PS_MODULE_DIR_.'cronjob',
```

Comment:
I'm quite worried by this list. It seems to be lost in the middle of the code, and we do not know why especially these modules should be concerned. What if another native/contributor module want to be added here?

Reply:
I agree. It's basically a whitelist of native modules that should be transformed to RTL automatically. I thought about setting it as a class property, but then there was the problem of the concatenation, availability of the constant at construction, and so on.

Originally, the idea was to transform _all_ modules. However, that could cause problems on modules that already were RTL-compatible, so this shortlist was made.

Modules can also be converted to RTL manually using a new button that appears on the module's configuration page (but they need to have one).

I think that eventually modules should be able to set up a property that specifies if the module should be converted to RTL automatically, manually, or not converted to RTL at all.

I don't believe we need to settle that right away though.

Reply:
If you need something more complex (i.e retuning an answer that depend on something), we can use the hook system for that.
Only the module registered to that hook specifying RTL compliant would be taken in account. :)

Reply:
Yeah, I think we can set that up later :)

Reply:
I think a property in module is enough. we can set a property to each module.
We can use $this->use_auto_rtl = true /false in module construction method,
then we can rtl it during Prestashop or module installation and when an RTL language is added.